src/backend/distributed/commands/utility_hook.c

```diff
@@ -224,10 +224,20 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 										   params, queryEnv, dest, completionTag);
 
 			StoredProcedureLevel -= 1;
+
+			if (InDelegatedProcedureCall && StoredProcedureLevel == 0)
+			{
+				InDelegatedProcedureCall = false;
```

Comment:
This happens here rather than in the commit hook because the procedure can commit.